### PR TITLE
[Core] Allow to run memray against dashboard / agent

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR allows to run memray against dashboard and agent. If the env var is configured, the memory profile dump will be created at `/tmp/ray/sessoin_latest/profile`. 

Here's the example output. The memray supports multiple output from stats, summary, tree, table, and flamegraph. 

Screen Shot 2022-10-14 at 10 49 10 AM

This PR will be used to debug memory leak of agent (and the dashboard in the future). 

https://github.com/ray-project/ray/issues/29199

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/services.py [hunk 1]
```diff
@@ -1306,6 +1353,7 @@ def start_api_server(
             f"--logging-rotate-backup-count={backup_count}",
             f"--gcs-address={gcs_address}",
         ]
+        print(command)
```
Comment (rickyyx) [inline python/ray/_private/services.py:1356]: is this intentional?

**Review by rickyyx — COMMENTED** [review]
Great stuff! Just a couple of nits and questions.

File: python/ray/_private/services.py [hunk 2]
```diff
@@ -91,6 +101,42 @@
 )
 
 
+def _build_python_executable_command(
```
[the remark this answers is not in the document]
Comment (rkooo567) [reply python/ray/_private/services.py:104]: Let me do with the minimal change first. I think we can definitely make this method more generic, but I'd like to merge this PR asap

**Review by rickyyx — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/gifs/eating-mrw-yyZRSvISN1vvW">GIPHY</a></div>

**Comment by rickyyx** [comment]
Seems there are still some tests failing.

**Pull request merged**